evm_loader/program/src/account/sysvar.rs

```diff
@@ -0,0 +1,145 @@
+use crate::error::Error;
+use crate::gasometer::LAMPORTS_PER_SIGNATURE;
+use ethnum::U256;
+use solana_program::program_error::ProgramError;
+use solana_program::{account_info::AccountInfo, pubkey::Pubkey, sysvar::instructions};
+use std::convert::From;
+use std::ops::Deref;
+use std::str::FromStr;
+
+// Because ComputeBudget program is not accessible through CPI, it's not a part of the standard
+// solana_program library crate. Thus, we have to hardcode a couple of constants.
+// The pubkey of the Compute Budget.
+const COMPUTE_BUDGET_ADDRESS: &str = "ComputeBudget111111111111111111111111111111";
+// The Compute Budget SetComputeUnitLimit instruction tag.
+const COMPUTE_UNIT_LIMIT_TAG: u8 = 0x2;
+// The Compute Budget SetComputeUnitPrice instruction tag.
+const COMPUTE_UNIT_PRICE_TAG: u8 = 0x3;
+// The default compute units limit for Solana transactions.
+const DEFAULT_COMPUTE_UNIT_LIMIT: u32 = 200_000;
+
+// The divisor in the conversion from priority fee in microlamports to priority fee per gas.
+// This divisor includes the tolerance_level := 0.9 which is an allowed discrepancy between
+// the actual priority fee per gas as paid by the Operator and the priority fee per gas
+// as set by the User in the transaction.
+const CONVERSION_DIVISOR: u64 = LAMPORTS_PER_SIGNATURE * 900_000;
+
+pub struct Sysvar<'a>(&'a AccountInfo<'a>);
+
+impl<'a> From<&Sysvar<'a>> for &'a AccountInfo<'a> {
+    fn from(f: &Sysvar<'a>) -> Self {
+        f.0
+    }
+}
+
+impl<'a> Sysvar<'a> {
+    pub fn from_account(info: &'a AccountInfo<'a>) -> Result<Self, ProgramError> {
+        if !instructions::check_id(info.key) {
+            return Err!(ProgramError::InvalidArgument; "Account {} - is not sysvar program", info.key);
+        }
+
+        Ok(Self(info))
+    }
+
+    /// Extracts the data about compute units from instructions within the current transaction.
+    /// Returns the pair of (`compute_budget_unit_limit`, `compute_budget_unit_price`)
+    /// N.B. the `compute_budget_unit_price` is denominated in micro Lamports.
+    pub fn get_compute_budget_priority_fee(&self) -> Result<(u32, u64), Error> {
+        let compute_budget_account_pubkey =
+            Pubkey::from_str(COMPUTE_BUDGET_ADDRESS).map_err(|_| {
+                Error::PriorityFeeParsingError("Invalid Compute budget address.".to_string())
+            })?;
+        // Intent is to check all the instructions before the current one.
```

Comment:
Thanks, checked. It seems that for transaction to be valid, the solana runtime first need to sanitize transaction:
- check that compute budget limit are set correctly (e.g. limit < 1.4M, heap < 256kb, etc)
- actually enforce those limit by creating SVM with specified values (heap size, CU limit, etc).

https://github.com/anza-xyz/agave/blob/70efe819bcf70476926f06c473f29015b3269536/runtime-transaction/src/runtime_transaction.rs#L89

So long story short, CB handling goes before actual transaction logic and indeed, **it's not required to have CB placed first in the instruction list**. I doubled checked it locally by placing CB instructions at the end.

On the other hand:
- proxy does exactly this,
- most of transactions I've seen are doing the same (it's quite logical),
- we have no other way, but to rely this fact (after I removed Sysvar and substituted it with the syscall `instruction::get_**processed**_sibling_instruction`, please see recent changes). 

The processed word is important here, it implies that CB instructions should go first. Otherwise syscall does not see CB instuction, I also checked it locally. 

What's your opinion?